src/ICSCore.sol

```diff
@@ -2,15 +2,16 @@
 pragma solidity ^0.8.28;
 
 import { IICS02Client } from "./interfaces/IICS02Client.sol";
+import { IICS04Channel } from "./interfaces/IICS04Channel.sol";
 import { Strings } from "@openzeppelin/utils/Strings.sol";
 import { IBCIdentifiers } from "./utils/IBCIdentifiers.sol";
 import { ILightClient } from "./interfaces/ILightClient.sol";
 import { IICS02ClientErrors } from "./errors/IICS02ClientErrors.sol";
 import { Ownable } from "@openzeppelin/access/Ownable.sol";
 
-contract ICS02Client is IICS02Client, IICS02ClientErrors, Ownable {
-    /// @dev clientId => counterpartyInfo
-    mapping(string clientId => CounterpartyInfo info) private counterpartyInfos;
+contract ICSCore is IICS02Client, IICS04Channel, IICS02ClientErrors, Ownable {
+    /// @dev channelId => counterpartyInfo
+    mapping(string channelId => Channel channel) private channels;
```

Comment:
CounterpartyInfo feels more descriptive ngl 😅. Because the same light client cannot have multiple channels on top in our implementation I think